docs/root/intro/arch_overview/data_sharing_between_filters.rst

```diff
@@ -0,0 +1,93 @@
+.. _arch_overview_data_sharing_between_filters:
+
+Envoy provides the following mechanisms for the transfer of configuration,
+metadata and per-request/connection state to, from and between filters, as
+well as to other core subsystems  (e.g., access logging).
+
+
+Static State
```

Comment:
It would be really A++ if you could ref link to the various proto fields for all of these options?